pkg/base1/cockpit.js

```diff
@@ -3461,85 +3502,85 @@ function full_scope(cockpit, $, po) {
-                    var timestamp;
-
-                    /* A meta message? */
-                    var message_len = message.length;
-                    if (message_len === undefined) {
-                        meta = message;
-                        timestamp = 0;
-                        if (meta.now && meta.timestamp)
-                            timestamp = meta.timestamp + ($.now() - meta.now);
-                        beg = Math.floor(timestamp / interval);
-                        callback(beg, meta, null, options_list[0]);
-
-                    /* A data message */
-                    } else if (meta) {
-
-                        /* Data decompression */
-                        for (i = 0; i < message_len; i++) {
-                            data = message[i];
-                            if (last) {
-                                data_len = data.length;
-                                last_len = last.length;
-                                for (j = 0; j < last_len; j++) {
-                                    dataj = data[j];
-                                    if (dataj === null || dataj === undefined) {
-                                        data[j] = last[j];
-                                    } else {
-                                        dataj_len = dataj.length;
-                                        if (dataj_len !== undefined) {
-                                            lastj = last[j];
-                                            lastj_len = last[j].length;
-                                            for (k = 0; k < dataj_len; k++) {
-                                                if (dataj[k] === null)
-                                                    dataj[k] = lastj[k];
-                                            }
-                                            for (; k < lastj_len; k++)
+                } else if (is_archive) {
+                    if (!self.archives) {
+                        self.archives = true;
+                        self.dispatchEvent('changed');
+                    }
+                }
+            });
+
+            channel.addEventListener("message", function(ev, payload) {
+                var message = JSON.parse(payload);
+
+                var data, data_len, last_len, dataj, dataj_len, lastj, lastj_len;
+                var i, j, k;
+                var timestamp;
+
+                /* A meta message? */
+                var message_len = message.length;
+                if (message_len === undefined) {
+                    meta = message;
+                    timestamp = 0;
+                    if (meta.now && meta.timestamp)
+                        timestamp = meta.timestamp + ($.now() - meta.now);
+                    beg = Math.floor(timestamp / interval);
+                    callback(beg, meta, null, options_list[0]);
+
+                /* A data message */
+                } else if (meta) {
+
+                    /* Data decompression */
```

Comment:
I know this is from previous code, but is there a bit more to write here? Either a description of what kind of decompression or even a link to the corresponding compression would be nice.